internal/cluster/finalizing_stages_test.go

```diff
@@ -0,0 +1,83 @@
+package cluster
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	. "github.com/onsi/ginkgo/extensions/table"
+	. "github.com/onsi/gomega"
+	"github.com/openshift/assisted-service/models"
+	"github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+)
+
+var _ = DescribeTable("finalizing stage timeouts",
+	func(stage models.FinalizingStage, operators []*models.MonitoredOperator, envValue string, expected time.Duration) {
+		if envValue != "" {
+			envKey := convertStageToEnvVar(stage)
+			_ = os.Setenv(envKey, envValue)
+			defer func() {
+				_ = os.Unsetenv(envKey)
+			}()
+		}
+		Expect(finalizingStageTimeout(stage, operators, logrus.New())).To(Equal(expected))
+	},
+	func() []TableEntry {
+		// Variables for test setup
+		var (
+			ret       []TableEntry
+			log       = logrus.New()
+			olmStages = []models.FinalizingStage{
+				models.FinalizingStageWaitingForOlmOperatorsCsvInitialization,
+				models.FinalizingStageWaitingForOlmOperatorsCsv,
+			}
+			nonOlmStages = funk.Subtract(finalizingStages, olmStages).([]models.FinalizingStage)
+			toSeconds    = func(d time.Duration) int64 {
+				return int64(d.Seconds())
+			}
+			operators = []*models.MonitoredOperator{
+				{
+					OperatorType:   models.OperatorTypeBuiltin,
+					TimeoutSeconds: toSeconds(22 * time.Hour),
+				},
+				{
+					OperatorType:   models.OperatorTypeOlm,
+					TimeoutSeconds: toSeconds(20 * time.Hour),
+				},
+				{
+					OperatorType:   models.OperatorTypeOlm,
+					TimeoutSeconds: toSeconds(21 * time.Hour),
+				},
+			}
+		)
+		// End of variables declaration section
+
+		// Test cases for all stages without operators
+		for _, stage := range finalizingStages {
+			defaultTimeout := finalizingStageDefaultTimeout(stage, log)
+			ret = append(ret,
+				Entry(fmt.Sprintf("%s without environment and without operators", stage), stage, nil, "", defaultTimeout))
+			ret = append(ret,
+				Entry(fmt.Sprintf("%s with environment and without operators", stage), stage, nil, "123m", 123*time.Minute))
+		}
+
+		// Test cases for non OLM stages with operators.  Should behave the same as if operators were not provided
+		for _, stage := range nonOlmStages {
+			defaultTimeout := finalizingStageDefaultTimeout(stage, log)
+			ret = append(ret,
+				Entry(fmt.Sprintf("%s without environment and with operators", stage), stage, operators, "", defaultTimeout))
```

Comment:
Can you update the test names to describe what is being tested and the expected result? You have the info in the comment but it would be better if it were in the test name as well.

For example this would could be `"%s without environment and with operators uses the default timeout"`

Reply:
OK